packages/node/src/lib/retry.spec.ts

```diff
@@ -0,0 +1,205 @@
+import nock from 'nock';
+import { Novu } from '../index';
+
+const BACKEND_URL = 'http://example.com';
+const TOPICS_PATH = '/v1/topics';
+const TRIGGER_PATH = '/v1/events/trigger';
+
+jest.setTimeout(15000);
+
+const allEqual = (arr: Array<string>) => arr.every((val) => val === arr[0]);
```

Comment:
I disagree with second sentence but I understand where problem is. Function `allEqual` (currently renamed to `hasAllEqual`) works well when our goal is to check if all items are equal (it iterates over all elements and compares them to the first element, it also compares first element with itself). This is the goal when we check in tests if idempotency-keys are equal between retries (in the scope of same request).

I agree this function is used in wrong way when we check if idempotency-keys are unique between requests - we expect `allEqual`  to return `false`. Negation of this function does not guarantee that all items are unique, so this expectation will pass in case of `[1,2,2,2,2]` scenario, but shouldn't.

I decided to let them coexist :)